### DISCUSSION TITLE
Cannot login to Meshcentral server

### DISCUSSION THREAD

**Discussion by Suncatcher** [Ylianst/MeshCentral#5782] [Q&A] [answered]
Probably very newbie question but I cannot log in to the server, the login page is not showed
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/e9aa061a-e15d-4e59-902a-fb213d497201)

When I open the page it just shows this logo and nothing else, nothing is clickable on the page. Systemctl shows service properly started.
I tried to start another instance of meshcentral manually, but got the same result just on another port

    [ec2-user@ip-XXXXXXXXX ~]$ node node_modules/meshcentral --debug web,webrelay
    MeshCentral HTTP redirection server running on port 82.
    MeshCentral v1.1.20, LAN mode.
    Server has no users, next new account will be site administrator.
    MeshCentral HTTPS server running on port 445.

Any suggestions?

**Comment by si458** [comment]
Restart ur EC2 instance and try again,
It's getting itself confused because its not running on the standard port 80/443 but port 82/445, so it thinks something else is running on the normal 80/443 ports

**Reply by Suncatcher** [reply]
restarted multiple times, also reinstalled, that didn't help. 

> It's getting itself confused

how it is getting confused? this manual run I did only to illustrate, I see it starts normally on the standard 443 SSL port via autostart
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/10683485-4e9a-4523-a733-172755278dd6)

Is there any way to troubleshoot why there is no login page? how can I check the logs

**Reply by si458** [reply]
Ok So it's starting to correctly and on the correct ports ok, so load up the login page and check the systemctl status again and see if u get any errors listed there
Also can u share ur config.json file? (Hide secret info)

**Reply by Suncatcher** [reply]
> so load up the login page and check the systemctl status again and see if u get any errors listed there

refreshing the page changes absolutely nothing, I see exactly the same systemd status output, no errors.
I open the bare IP of the EC2 instance on SSL port. Is there any specific URL to open the login page or it is always domain root?

Giving my config.json

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "_cert": "myserver.mydomain.com",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
```

Sharing my config.json as is, as I didn't alter it, so it hardly contains any secret info.

**Reply by si458** [reply]
Try the following.
uncomment `_cert` to `cert` and enter your domain name,
Also uncomment letsencrypt too `_letsencrypt` to `letsencrypt`
And enter ur domain name into names, and set ur email into email
Then restart ur ec2 and check the output of systemctl status again

**Reply by si458** [reply]
You are also missing a closing bracket at the end of ur json? Bad copy+paste maybe? Check the json is valid with something like jsonlint.com

**Reply by Suncatcher** [reply]
> You are also missing a closing bracket at the end of ur json? Bad copy+paste maybe? Check the json is valid with something like jsonlint.com

no, the JSON was valid, I checked it via jsonlint.com. And how could it be not valid if I never changed it?

I did all your suggestions and changed config.json to this one

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "XXXXXXXXXXXXXX.us-west-1.compute.amazonaws.com",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXXXXXXXXXX@outlook.com",
    "names": "ec2-XXXXXXXXXX.us-west-1.compute.amazonaws.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```
and still the same error, empty non-clickable page. Though, the systems output slightly changed after your suggestions and now shows this
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/2086fcc6-58a1-46fb-98e9-c77f4ec68c24)
Looks like previously it was in LAN mode and now starts in WAN, which is more correct in my case as the machine (agent) I want to connect to will be not in my LAN but behind firewall.

**Reply by si458** [reply]
OK try this, use sudo for all
1. `systemctl stop meshcentral`
2. `cd meshcentralfolder`
3. `rm -rfv node_modules`
4. `cat package.json` - let me have output/reply
5. Make sure step 4 contains meshcentral in the dependencies
6. `npm install`
7. `systemctl start meshcentral`

**Reply by si458** [reply]
Also what's the layout of your meshcentral folder? It should contain node_modules, meshcentral-data, etc... `la -lah`

**Reply by Suncatcher** [reply]
> cat package.json - let me have output/reply

```
{
  "dependencies": {
    "acme-client": "4.2.5",
    "meshcentral": "^1.1.20",
    "otplib": "10.2.3"
  }
}
```

> Also what's the layout of your meshcentral folder?

```
total 172K
drwx------. 8 ec2-user ec2-user  16K Feb 13 03:33 .
drwxr-xr-x. 3 root     root       22 Feb  8 07:43 ..
-rw-------. 1 ec2-user ec2-user 1.4K Feb 12 03:51 .bash_history
-rw-r--r--. 1 ec2-user ec2-user   18 Jan 28  2023 .bash_logout
-rw-r--r--. 1 ec2-user ec2-user  141 Jan 28  2023 .bash_profile
-rw-r--r--. 1 ec2-user ec2-user  689 Feb  8 07:50 .bashrc
-rw-------. 1 ec2-user ec2-user   20 Feb  8 08:12 .lesshst
drwxr-xr-x. 4 ec2-user ec2-user   72 Feb 12 03:40 .npm
drwxr-xr-x. 5 ec2-user ec2-user  102 Feb  8 07:50 .nvm
drwx------. 2 ec2-user ec2-user   29 Feb  8 07:43 .ssh
-rw-------. 1 ec2-user ec2-user  13K Feb 12 03:51 .viminfo
drwxr-xr-x. 2 ec2-user ec2-user  16K Feb 12 04:40 meshcentral-backups
drwxr-xr-x. 4 ec2-user ec2-user  16K Feb 12 03:51 meshcentral-data
drwxr-xr-x. 2 ec2-user ec2-user    6 Feb  8 08:07 meshcentral-files
-rw-r--r--. 1 ec2-user ec2-user  66K Feb 12 03:40 package-lock.json
-rw-r--r--. 1 ec2-user ec2-user  109 Feb 12 03:40 package.json
-rwxr-xr-x. 1 ec2-user ec2-user   43 Feb  8 07:50 restart
-rwxr-xr-x. 1 ec2-user ec2-user   41 Feb  8 07:50 start
-rwxr-xr-x. 1 ec2-user ec2-user   40 Feb  8 07:50 stop
-rwxr-xr-x. 1 ec2-user ec2-user  106 Feb  8 07:50 update
```
this is the layout after removal of node_modules on the step 3. My meshcentral was installed in user folder (~), I was using this script, it is recommended by Meshcentral, yeah? [https://github.com/Ylianst/MeshCentralSite/blob/main/scripts/mc-aws-linux2.sh](https://github.com/Ylianst/MeshCentralSite/blob/main/scripts/mc-aws-linux2.sh)

Did all the steps you mentioned, after step 6 I saw

```
[ec2-user@ip-XXXXXXXXXXX ~]$ npm install

added 170 packages, and audited 171 packages in 9s

18 packages are looking for funding
  run `npm fund` for details

2 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
```

And after step 7 I checked the URL and the same issue, just blind logo page, nothing else.
Any more suggestions?

**Reply by si458** [reply]
Try a clean ec2 instance, start completely afresh,  only option I have left I'm afraid

**Reply by si458** [reply]
Just a thought, what version of nodejs is running? `node -version` 
You might need to update to node 18/20?

**Reply by Suncatcher** [reply]
> You might need to update to node 18/20?

v21.6.1

so v21 is not supported yet? Though I believe I had the same error on the previous version and I already was recreating the EC2 instance from scratch (full reinstall).
so you are out of ideas. No single suggestion how to troubleshoot this? The problem seems to be very trivial though my node knowledge is rather limited to find it.
Why the certificate is shown as invalid? Could it be the key to the problem?

**Reply by si458** [reply]
I dont think nodejs version has any issue unless its less than 16. But u could try installing 20? In theory the ec2 script has been updated recently to install 16 but will chsnge it to install 20 later today

As for The certificate showing invalid this is because meshcentral creates a self signed one by default, but should still work.

However To use a valid one you need to uncomment the `_letsencrypt` section and enter ur email address and the same dns name in there, set production to true, and restart meshcentral, then wait 2 mins and refresh the web page

**Reply by Suncatcher** [reply]
wow, the problem was in the browser, lol. I just tried to open it from the mobile browser and the login page is showed normally.
I knew my Chrome desktop was cutting scripts but I didn't know it was doing it so aggressively, haha.
Many thanks for trying to help me! I am so grateful.
however your suggestion regarding the certificate didn't work. I uncommented all the attributes and filled it per your recommendation but it still throwing the error
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/b61f0fda-de77-486b-8b18-3721f7fbfbf2)
here is my let's encrypt section config
```
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXXXXXXXXXX@outlook.com",
    "names": "ec2-XX-XX-XXX-XX.us-west-1.compute.amazonaws.com",
    "skipChallengeVerification": true,
    "production": false
  }
```
am I missing something?

**Reply by si458** [reply]
from the looks of the screenshot, you are accessing the web panel by its IP address `https://13.`
you need to access it via a DNS name not an ip address!
so the DNS name in `letsencrypt->names` needs to match `settings->cert`!
you also need to set `letsencrypt->production` to `true` to get a VALID SSL,
but again YOU MUST HAVE A DNS THATS VALID AND POINTS DIRECTLY TO YOUR MESHCENTRAL SERVER!

**Comment by Suncatcher** [comment]
> you need to access it via a DNS name not an ip address!

the problem is DNS access doesn't work, it's always stuck at this 
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/9f752fe8-5f58-412b-9d43-d38aa1497b24)
and then fails
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/42c9e986-e8f3-41db-b1c6-b4fb8d1787f4)

**Reply by si458** [reply]
what about typing `https://ec2-XX-XX-XXX-XX.us-west-1.compute.amazonaws.com` rather than `ec2-XX-XX-XXX-XX.us-west-1.compute.amazonaws.com` ?
it that works, you might need to check you have port 80 open
also check you have the public DNS name and not the internal one? https://docs.aws.amazon.com/vpc/latest/userguide/vpc-dns.html#vpc-dns-viewing

**Comment by Suncatcher** [comment]
My full config.json

```
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "cert": "ec2-00-00-00-00.us-west-1.compute.amazonaws.com",
    "WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "XXXXXXXXXXX@outlook.com",
    "names": "ec2-00-00-00-00.us-west-1.compute.amazonaws.com",
    "skipChallengeVerification": true,
    "production": true
  }
}
```

The parameters `names` and `cert` are the same and equal to `ec2-00-00-00-00`. I obfuscated it here but this value is real in config.
It opens the same insecure in the desktop, but desktop is a bad example for the abovementioned issues, and on the phone it is not reachable at all
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/c934d0c2-963e-4a01-866d-b427242a0630)

**Reply by si458** [reply]
this to me screams something is blocking the dns names?
whats ur ec2 firewall rules like? port 80+443 wide open to 0.0.0.0 ?
does the ec2 have ipv6? do you have ipv6?

**Comment by Suncatcher** [comment]
> what about typing https://ec2-XX-XX-XXX-XX.us-west-1.compute.amazonaws.com rather than ec2-XX-XX-XXX-XX.us-west-1.compute.amazonaws.com

okay, you were right. My mobile browser is defaulting to HTTP. When i explicitly opened HTTPS it worked by DNS.
But again insecure, certificate error
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/bddf02a3-924c-4efe-8bf3-becd1d485b57)
Any suggestion?

**Reply by si458** [reply]
whats ur ec2 firewall rules like? port 80+443 wide open to 0.0.0.0 ?
it would explain why the SSL is invalid because letsencrypt needs port 80 open for them to verify you for the SSL

**Comment by Suncatcher** [comment]
> whats ur ec2 firewall rules like? port 80+443 wide open to 0.0.0.0 ?

yup
![image](https://github.com/Ylianst/MeshCentral/assets/6388034/7ca16699-1a19-4792-b56d-c6ac278f84e1)

**Reply by si458** [reply]
check ur screenshot... you dont have port 80 open...
80=http, 443=https
you also need 4433 too BUT ONLY IF you want to use intel amt 👍

**Reply by Suncatcher** [reply]
yes, I was not completely true. Now I added the 80 port, the error is not gone. should I restart the instance and/or restart the meshcentral?
I was thinking security groups rules are applied instantly

**Reply by si458** [reply]
yes restart meshcentral, wait about 2 mins, then fresh the webpage and it should be secured now

**Reply by Suncatcher** [reply]
> yes restart meshcentral, wait about 2 mins, then fresh the webpage and it should be secured now

restarted the mesh, that didn't help. Then restarted the instance, that didn't help either. The certificate error is still showing up.
Any other suggestions?

**Reply by si458** [reply]
whats the output when you run meshcentral?
`sudo systemctl status meshcentral` OR stop the service and run manually `node node_modules/meshcentral --debug`
is the any extra lines? any errors? any warnings?

**Comment by Suncatcher** [comment]
> whats the output when you run meshcentral?
```
[ec2-user@ip-XXX-XX-XX-XX ~]$ sudo systemctl status meshcentral
● meshcentral.service - MeshCentral Server
     Loaded: loaded (/etc/systemd/system/meshcentral.service; enabled; preset: disabled)
     Active: active (running) since Thu 2024-02-15 20:05:46 UTC; 50min ago
   Main PID: 1423 (node)
      Tasks: 29 (limit: 407)
     Memory: 132.9M
        CPU: 8.210s
     CGroup: /system.slice/meshcentral.service
             ├─1423 /home/ec2-user/.nvm/versions/node/v21.6.1/bin/node /home/ec2-user/node_modules/meshcentral
             └─1482 /home/ec2-user/.nvm/versions/node/v21.6.1/bin/node /home/ec2-user/node_modules/meshcentral --launch>

Feb 15 20:05:46 ip-XXX-XX-XX-XX.us-west-1.compute.internal systemd[1]: Started meshcentral.service - MeshCentral Serve>
Feb 15 20:05:49 ip-XXX-XX-XX-XX.us-west-1.compute.internal node[1423]: MeshCentral HTTP redirection server running on >
Feb 15 20:05:49 ip-XXX-XX-XX-XX.us-west-1.compute.internal node[1423]: MeshCentral v1.1.20, WAN mode.
Feb 15 20:05:54 ip-XXX-XX-XX-XX.us-west-1.compute.internal node[1423]: MeshCentral Intel(R) AMT server running on ec2->
Feb 15 20:05:54 ip-XXX-XX-XX-XX.us-west-1.compute.internal node[1423]: MeshCentral HTTPS server running on ec2-XX-XX-2>
```
> and run manually node node_modules/meshcentral --debug
```
MAIN: Core module windows-amt is 610162 bytes.
MAIN: Core module linux-amt is 556461 bytes.
MAIN: Core module linux-noamt is 455925 bytes.
MAIN: Core module windows-recovery is 137824 bytes.
MAIN: Core module linux-recovery is 82212 bytes.
MAIN: Core module windows-agentrecovery is 61977 bytes.
MAIN: Core module linux-agentrecovery is 6365 bytes.
MAIN: Core module windows-tiny is 6305 bytes.
MAIN: Core module linux-tiny is 6305 bytes.
MeshCentral HTTP redirection server running on port 80.
AUTHLOG: Server listening on 0.0.0.0 port 80.
CERT: LE: Getting certs from local store (Production)
CERT: LE: No certificate files found
MeshCentral v1.1.20, WAN mode.
DISPATCH: AddEventDispatch [ 'server-shareremove' ]
DISPATCH: AddEventDispatch [ '*' ]
DISPATCH: DispatchEvent [ '*' ]
MAIN: Server started
MeshCentral Intel(R) AMT server running on ec2-XXX-XX-XX-XX.us-west-1.compute.amazonaws.com:4433.
AUTHLOG: Server listening on 0.0.0.0 port 4433.
AUTHLOG: Server listening on 0.0.0.0 port 443.
MeshCentral HTTPS server running on ec2-XXX-XX-XX-XX.us-west-1.compute.amazonaws.com:443.
CERT: LE: Got no certificates, asking for one now.
CERT: LE: Generating private key...
CERT: LE: Setting up ACME client...
CERT: LE: Creating certificate request...
CERT: LE: Requesting certificate from Let's Encrypt...
CERT: LE: Failed to obtain certificate: Error creating new order :: Cannot issue for "ec2-XXX-XX-XX-XX.us-west-1.compute.amazonaws.com": The ACME server refuses to issue a certificate for this domain name, because it is forbidden by policy
```

Forbidden by which policy?

**Reply by Suncatcher** [reply]
> you also need 4433 too BUT ONLY IF you want to use intel amt

I see the server is running on 4433 port also, but I'm on t4g-type instance which is ARM-based, so this is not relevant to me, it is not intel. How do I disable it and why it is enabled by default?

**Reply by si458** [reply]
Your problem with the ssl is explained in the very last line!? 
Letsencrypt forbid creating ssl certs with the aws domain name
So u need another dns name
Also it doesn't matter what processor u run meshcentral on, the amt is coded eith the node so its safe to use arm
You can disable by setting mpsport to 0 under settings in config.json if I remember?

**Reply by Suncatcher** [reply]
> Letsencrypt forbid creating ssl certs with the aws domain name

wow, so bad. You didn't know that? We could save such a lot of time troubleshooting this, lol
anyway, many thanks for your support, I really appreciate that.
I will consider making own domain name for mesh server.

**Reply by si458** [reply, marked as answer]
No worries no harm, and No I didn't realise letsencrypt forbid using the aws domain! 
But I guess they are right to forbid it tho as the ip addresses/Ec2 instances can be chopped and changed very quickly!
I personally use duckdns.org for a free ddns for my home stuff 👍 
I think they even have scripts u can setup if ya ec2 instance ip address changes haha

**Reply by Suncatcher** [reply]
I set up a domain, linked it and changed the config like you said and no error anymore! Everything works beautifully. Kudos to you!
